File: C++/Recursion/TowerOfHanoi.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution{
    public:
    long long count=0;
    //function for tower of hanoi
    long long toh(int N, int from, int to, int aux) {
       if(N==1)
       {
           cout<<"move disk "<<N<<" "<<"from rod "<<from<<" "<<"to rod "<<to<<endl;
           count++;
           return count;
       }
       toh(N-1,from,aux,to);
       cout<<"move disk "<<N<<" "<<"from rod "<<from<<" "<<"to rod "<<to<<endl;
           count++;
        toh(N-1,aux,to,from);
         return count;
    }

};
int main() {

    int T;
    cin >> T;//testcases
    while (T--) {
        
        int N;
        cin >> N;//taking input N
        
        //calling toh() function
        Solution ob;
        
        cout << ob.toh(N, 1, 3, 2) << endl;
    }
    return 0;
}

File: Java/TowerOfHanoi.java
package daa.assignment;

public class TowerOfHanoi {
    
    public static void main(String[] args) {
        towerOfHanoi(4, 'A', 'B', 'C');
    }
    
    public static void towerOfHanoi(int n, char from, char to, char aux) {
        if (n == 1) {
            System.out.println("Move disk 1 from " + from + " to " + to);
            
        }else {
            towerOfHanoi(n - 1, from, aux, to);
        
            System.out.println("Move disk " + n + " from " + from + " to " + to);
        
            towerOfHanoi(n - 1, aux, to, from);
        }
    }
    
}
